libs/server/Resp/Vector/RespServerSessionVectors.cs

```diff
@@ -1004,11 +1004,47 @@ private bool NetworkVINFO<TGarnetApi>(ref TGarnetApi storageApi)
                 return AbortWithErrorMessage("ERR Vector Set (preview) commands are not enabled");
             }
 
-            // TODO: implement!
+            if (parseState.Count != 1)
+            {
+                return AbortWithWrongNumberOfArguments("VINFO");
+            }
 
-            while (!RespWriteUtils.TryWriteDirect(CmdStrings.RESP_OK, ref dcurr, dend))
-                SendAndReset();
+            var key = parseState.GetArgSliceByRef(0);
+            var res = storageApi.VectorSetInfo(key, out VectorQuantType quantType, out uint vectorDimensions, out uint reducedDimensions, out uint buildExplorationFactor, out uint numLinks, out long size);
+            if (res == GarnetStatus.NOTFOUND)
+            {
+                WriteNull();
+                return true;
+            }
+            
+            if (res == GarnetStatus.WRONGTYPE)
+            {
+                WriteError("ERR Not a Vector Set"u8);
+                return true;
+            }
 
+            string quantTypeStr = quantType switch
```

Comment:
nit: use type inference (ie. `var`) in the Garnet project.  This differs from some other coding standards, but its preferred here.